File: pro1.cpp
#include <vector>
#include <algorithm>
#include <string>
#include <list>
#include <queue>
#include <stack>
#include <set>
#include <map>
#include <unordered_map>
#include <unordered_set>
using namespace std;

class Solution {
public:
    int similarPairs(vector<string>& words) {
        map<set<char>, int> dics;
        for (auto& word : words) {
            set<char> s;
            for (auto& ch : word) {
                s.emplace(ch);
            }
			dics[s] += 1;
        }
		int ans = 0;
		for (auto& elem : dics) {
			if (elem.second > 1) ans += elem.second * (elem.second - 1) / 2;
		}
		return ans;
    }
};

File: pro1.py
from collections import defaultdict
from typing import List


class Solution:
    def similarPairs(self, words: List[str]) -> int:
        dicts = defaultdict(int)
        for word in words:
            dic = [0 for i in range(26)]
            for ch in word: dic[ord(ch) - ord("a")] = 1
            dicts[dic] += 1
        ans = 0
        keys = dicts.keys()
        for key in keys:
            if (dicts[key] > 1): ans += dicts[key] * (dicts[key] - 1) // 2
        return ans